### ISSUE TITLE
[Affiliate] Persist user's referral info on redux

### ISSUE THREAD

**Issue by alfetopito** [gnosis/cowswap#945]
# Summary

Persist the referral info on redux store

# Requirements

- Implement redux store boilerplate. Add, update, etc. Whatever is needed with hooks/actions/etc
- ~Store both the JSON doc and the hash~
- ~Proposed structure (feel free to propose a different one)~ See new proposed structure at the end
```json
{
  metadata: <the json metadata>,
  hash: "<the metadata doc hash>"
}
```

Depends on https://github.com/gnosis/cowswap/issues/850
Update state created on that task to add referral info (metadata doc, hash)
~Depends on #941~
~Use the types created there~
Update the types to the ones proposed below

# Out of scope

- Add hash to order

# Additional context

- Wireframes: https://www.figma.com/file/P0AIwKkIUpEZtLukkCUM94/CowSwap_-Affiliate?node-id=48%3A224
- Design doc: https://docs.google.com/document/d/1PsdWn9WPFf2BNeoJNtClyra8jNvWk_bJ-g33TbpUf8s/edit#

## Update 2021/07/19

Added reference to #850 and #941

## Update 2021/08/03

As pointed out by Ramiro below, updated the task based on changes from https://github.com/gnosis/cowswap/issues/944#issuecomment-890250294

New proposed type definition for referral info:

```typescript

type AppDataInfo = {
  referralAddress?: string,
  appDataHash?: string,
}
```

**Comment by ramirotw** [comment]
@alfetopito @anxolin this ticket should be updated with the latest spec changes from https://github.com/gnosis/cowswap/issues/944#issuecomment-890250294. Only `appDataHash` should be persisted.
